pydicom_seg/reader.py

```diff
@@ -167,8 +167,14 @@ def read(self, dataset: pydicom.Dataset) -> SegmentReadResult:
 
             # Iterate over all frames and check for referenced segment number
             for frame_idx, pffg in enumerate(dataset.PerFrameFunctionalGroupsSequence):
-                if segment_number != pffg.SegmentIdentificationSequence[0].ReferencedSegmentNumber:
-                    continue
+                if hasattr(pffg, 'SegmentIdentificationSequence'):
+                    sis = pffg.SegmentIdentificationSequence[0]
+                elif hasattr(dcm.SharedFunctionalGroupsSequence[0], 'SegmentIdentificationSequence'):
+                    sis = dcm.SharedFunctionalGroupsSequence[0].SegmentIdentificationSequence[0]
```

Comment:
I would move this part up, out of the loop. The shared `SegmentIdentificationSequence` is constant and therefore should be extracted only once, if it is not present then set it to `None`.